llarp/win32/win32_intrnl.c

```diff
@@ -1,572 +1,676 @@
+
+  if(NT_SUCCESS(status))
+  {
+    for(i = 0; i < numInterfaces; i++)
+    {
+      if(!strcmp((PCHAR)ifInfo[i].if_info.ent.if_descr, name))
+      {
+        memcpy(info, &ifInfo[i], sizeof(*info));
+        break;
+      }
+    }
+
+    HeapFree(GetProcessHeap(), 0, ifInfo);
+
+    return i < numInterfaces ? 0 : 0xc0000001;
+  }
+
+  return status;
+}
+
+NTSTATUS
+getInterfaceInfoByIndex(HANDLE tcpFile, DWORD index, IFInfo *info)
+{
+  IFInfo *ifInfo;
+  DWORD numInterfaces;
+  NTSTATUS status = getInterfaceInfoSet(tcpFile, &ifInfo, &numInterfaces);
+  unsigned i;
+
+  if(NT_SUCCESS(status))
+  {
+    for(i = 0; i < numInterfaces; i++)
+    {
+      if(ifInfo[i].if_info.ent.if_index == index)
+      {
+        memcpy(info, &ifInfo[i], sizeof(*info));
+        break;
+      }
+    }
+
+    HeapFree(GetProcessHeap(), 0, ifInfo);
+
+    return i < numInterfaces ? 0 : 0xc0000001;
+  }
+
+  return status;
+}
+
+NTSTATUS
+getIPAddrEntryForIf(HANDLE tcpFile, char *name, DWORD index, IFInfo *ifInfo)
+{
+  NTSTATUS status = name ? getInterfaceInfoByName(tcpFile, name, ifInfo)
+                         : getInterfaceInfoByIndex(tcpFile, index, ifInfo);
+
+  if(!NT_SUCCESS(status))
+  {
+    fprintf(stderr, "getIPAddrEntryForIf returning %lx\n", status);
+  }
+
+  return status;
+}
+
+InterfaceIndexTable *
+getInterfaceIndexTableInt(BOOL nonLoopbackOnly)
+{
+  DWORD numInterfaces, curInterface = 0;
+  unsigned i;
+  IFInfo *ifInfo;
+  InterfaceIndexTable *ret = 0;
+  HANDLE tcpFile;
+  NTSTATUS status = openTcpFile(&tcpFile, FILE_READ_DATA);
+
+  ifInfo = NULL;
+
+  if(NT_SUCCESS(status))
+  {
+    status = getInterfaceInfoSet(tcpFile, &ifInfo, &numInterfaces);
+
+    fprintf(stderr, "InterfaceInfoSet: %08lx, %04lx:%08lx\n", status,
+            ifInfo->entity_id.tei_entity, ifInfo->entity_id.tei_instance);
+
+    if(NT_SUCCESS(status))
+    {
+      ret = (InterfaceIndexTable *)calloc(
+          1, sizeof(InterfaceIndexTable) + (numInterfaces - 1) * sizeof(DWORD));
+
+      if(ret)
+      {
+        ret->numAllocated = numInterfaces;
+        fprintf(stderr, "NumInterfaces = %ld\n", numInterfaces);
+
+        for(i = 0; i < numInterfaces; i++)
+        {
+          fprintf(stderr, "Examining interface %d\n", i);
+          if(!nonLoopbackOnly || !isLoopback(tcpFile, &ifInfo[i].entity_id))
+          {
+            fprintf(stderr, "Interface %d matches (%ld)\n", i, curInterface);
+            ret->indexes[curInterface++] = ifInfo[i].if_info.ent.if_index;
+          }
+        }
+
+        ret->numIndexes = curInterface;
+      }
+
+      tdiFreeThingSet(ifInfo);
+    }
+    closeTcpFile(tcpFile);
+  }
+
+  return ret;
+}
+
+InterfaceIndexTable *
+getInterfaceIndexTable(void)
+{
+  return getInterfaceIndexTableInt(FALSE);
+}
+
+#endif
+
+// there's probably an use case for a _newer_ implementation
+// of pthread_setname_np(3), in fact, I may just merge _this_
+// upstream...
+#ifdef _MSC_VER

```

Comment:
looks like the line endings changed in this file, was that intentional?

Reply:
yeah, i was using visual studio, then git changed it to UNIX line ends